async_simple/coro/Collect.h

```diff
@@ -122,6 +128,83 @@ struct CollectAnyAwaiter {
     std::shared_ptr<ResultType> _result;
 };
 
+template <template <typename> typename LazyType, typename... Ts>
+struct CollectAnyVariadicAwaiter {
+    using ResultType = std::variant<Try<Ts>...>;
+    using InputType = std::tuple<LazyType<Ts>...>;
+
+    CollectAnyVariadicAwaiter(LazyType<Ts>&&... inputs)
+        : _input(std::make_unique<InputType>(std::move(inputs)...)),
+          _result(nullptr) {}
+
+    CollectAnyVariadicAwaiter(const CollectAnyVariadicAwaiter&) = delete;
+    CollectAnyVariadicAwaiter& operator=(const CollectAnyVariadicAwaiter&) =
+        delete;
+    CollectAnyVariadicAwaiter(CollectAnyVariadicAwaiter&& other)
+        : _input(std::move(other._input)), _result(std::move(other._result)) {}
+
+    bool await_ready() const noexcept {
+        return std::tuple_size<InputType>() == 0 ||
+               (_result && _result->has_value());
+    }
+
+    template <size_t... index>
+    void await_suspend_impl(std::index_sequence<index...>,
+                            std::coroutine_handle<> continuation) {
+        auto promise_type =
+            std::coroutine_handle<LazyPromiseBase>::from_address(
+                continuation.address())
+                .promise();
+        auto executor = promise_type._executor;
+
+        auto input = std::move(_input);
+        // Make local copies to shared_ptr to avoid deleting objects too early
+        // if any coroutine finishes before this function.
+        auto result = std::make_shared<std::optional<ResultType>>();
+        auto event =
+            std::make_shared<detail::CountEvent>(std::tuple_size<InputType>());
+
+        _result = result;
+
+        (
+            [&]() {
```

Comment:
We'd better align this.
```suggestion
        ( [&]() {
```